### ISSUE TITLE
Many many "tls: A TLS packet with unexpected length was received." warnings when playing Youtube videos with subtitles

### ISSUE THREAD

**Issue by sebma** [mpv-player/mpv#4220]
### mpv version and platform
mpv 0.24 on Ubuntu.

Hi,

I get many warning when playing youtube videos with subtitles : 
```
$ mpv --no-resume-playback https://www.youtube.com/watch?v=kiQa3DiiVys
Playing: https://www.youtube.com/watch?v=kiQa3DiiVys
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
[ffmpeg] tls: A TLS packet with unexpected length was received.
 (+) Video --vid=1 (*) (h264)
 (+) Audio --aid=1 --alang=eng (*) 'DASH audio' (opus) (external)
     Subs  --sid=1 --slang=xh 'vtt' (webvtt) (external)
     Subs  --sid=2 --slang=yo 'vtt' (webvtt) (external)
     Subs  --sid=3 --slang=th 'vtt' (webvtt) (external)
     Subs  --sid=4 --slang=sw 'vtt' (webvtt) (external)
     Subs  --sid=5 --slang=ar 'vtt' (webvtt) (external)
     Subs  --sid=6 --slang=sd 'vtt' (webvtt) (external)
     Subs  --sid=7 --slang=bn 'vtt' (webvtt) (external)
     Subs  --sid=8 --slang=eo 'vtt' (webvtt) (external)
     Subs  --sid=9 --slang=is 'vtt' (webvtt) (external)
     Subs  --sid=10 --slang=ig 'vtt' (webvtt) (external)
     Subs  --sid=11 --slang=cs 'vtt' (webvtt) (external)
     Subs  --sid=12 --slang=es 'vtt' (webvtt) (external)
     Subs  --sid=13 --slang=mg 'vtt' (webvtt) (external)
     Subs  --sid=14 --slang=lo 'vtt' (webvtt) (external)
     Subs  --sid=15 --slang=ps 'vtt' (webvtt) (external)
     Subs  --sid=16 --slang=no 'vtt' (webvtt) (external)
     Subs  --sid=17 --slang=bg 'vtt' (webvtt) (external)
     Subs  --sid=18 --slang=haw 'vtt' (webvtt) (external)
     Subs  --sid=19 --slang=ro 'vtt' (webvtt) (external)
     Subs  --sid=20 --slang=iw 'vtt' (webvtt) (external)
     Subs  --sid=21 --slang=uk 'vtt' (webvtt) (external)
     Subs  --sid=22 --slang=st 'vtt' (webvtt) (external)
     Subs  --sid=23 --slang=bs 'vtt' (webvtt) (external)
     Subs  --sid=24 --slang=mk 'vtt' (webvtt) (external)
     Subs  --sid=25 --slang=tg 'vtt' (webvtt) (external)
     Subs  --sid=26 --slang=ceb 'vtt' (webvtt) (external)
     Subs  --sid=27 --slang=gl 'vtt' (webvtt) (external)
     Subs  --sid=28 --slang=lt 'vtt' (webvtt) (external)
     Subs  --sid=29 --slang=el 'vtt' (webvtt) (external)
     Subs  --sid=30 --slang=pt 'vtt' (webvtt) (external)
     Subs  --sid=31 --slang=fil 'vtt' (webvtt) (external)
     Subs  --sid=32 --slang=ku 'vtt' (webvtt) (external)
     Subs  --sid=33 --slang=lb 'vtt' (webvtt) (external)
     Subs  --sid=34 --slang=nl 'vtt' (webvtt) (external)
     Subs  --sid=35 --slang=ml 'vtt' (webvtt) (external)
     Subs  --sid=36 --slang=sl 'vtt' (webvtt) (external)
     Subs  --sid=37 --slang=it 'vtt' (webvtt) (external)
     Subs  --sid=38 --slang=ht 'vtt' (webvtt) (external)
     Subs  --sid=39 --slang=sm 'vtt' (webvtt) (external)
     Subs  --sid=40 --slang=pl 'vtt' (webvtt) (external)
     Subs  --sid=41 --slang=hr 'vtt' (webvtt) (external)
     Subs  --sid=42 --slang=so 'vtt' (webvtt) (external)
     Subs  --sid=43 --slang=zu 'vtt' (webvtt) (external)
     Subs  --sid=44 --slang=si 'vtt' (webvtt) (external)
     Subs  --sid=45 --slang=sk 'vtt' (webvtt) (external)
     Subs  --sid=46 --slang=km 'vtt' (webvtt) (external)
     Subs  --sid=47 --slang=sq 'vtt' (webvtt) (external)
     Subs  --sid=48 --slang=kn 'vtt' (webvtt) (external)
     Subs  --sid=49 --slang=ko 'vtt' (webvtt) (external)
     Subs  --sid=50 --slang=sr 'vtt' (webvtt) (external)
     Subs  --sid=51 --slang=ru 'vtt' (webvtt) (external)
     Subs  --sid=52 --slang=su 'vtt' (webvtt) (external)
     Subs  --sid=53 --slang=kk 'vtt' (webvtt) (external)
     Subs  --sid=54 --slang=id 'vtt' (webvtt) (external)
     Subs  --sid=55 --slang=ny 'vtt' (webvtt) (external)
     Subs  --sid=56 --slang=zh-Hant 'vtt' (webvtt) (external)
     Subs  --sid=57 --slang=fy 'vtt' (webvtt) (external)
     Subs  --sid=58 --slang=fa 'vtt' (webvtt) (external)
     Subs  --sid=59 --slang=ca 'vtt' (webvtt) (external)
     Subs  --sid=60 --slang=da 'vtt' (webvtt) (external)
     Subs  --sid=61 --slang=ky 'vtt' (webvtt) (external)
     Subs  --sid=62 --slang=hi 'vtt' (webvtt) (external)
     Subs  --sid=63 --slang=ur 'vtt' (webvtt) (external)
     Subs  --sid=64 --slang=fi 'vtt' (webvtt) (external)
     Subs  --sid=65 --slang=af 'vtt' (webvtt) (external)
     Subs  --sid=66 --slang=eu 'vtt' (webvtt) (external)
     Subs  --sid=67 --slang=ka 'vtt' (webvtt) (external)
     Subs  --sid=68 --slang=la 'vtt' (webvtt) (external)
     Subs  --sid=69 --slang=de 'vtt' (webvtt) (external)
     Subs  --sid=70 --slang=ja 'vtt' (webvtt) (external)
     Subs  --sid=71 --slang=be 'vtt' (webvtt) (external)
     Subs  --sid=72 --slang=gu 'vtt' (webvtt) (external)
     Subs  --sid=73 --slang=et 'vtt' (webvtt) (external)
     Subs  --sid=74 --slang=vi 'vtt' (webvtt) (external)
     Subs  --sid=75 --slang=yi 'vtt' (webvtt) (external)
     Subs  --sid=76 --slang=ta 'vtt' (webvtt) (external)
     Subs  --sid=77 --slang=tr 'vtt' (webvtt) (external)
     Subs  --sid=78 --slang=hmn 'vtt' (webvtt) (external)
     Subs  --sid=79 --slang=co 'vtt' (webvtt) (external)
     Subs  --sid=80 --slang=pa 'vtt' (webvtt) (external)
     Subs  --sid=81 --slang=sv 'vtt' (webvtt) (external)
     Subs  --sid=82 --slang=uz 'vtt' (webvtt) (external)
     Subs  --sid=83 --slang=ne 'vtt' (webvtt) (external)
     Subs  --sid=84 --slang=my 'vtt' (webvtt) (external)
     Subs  --sid=85 --slang=te 'vtt' (webvtt) (external)
     Subs  --sid=86 --slang=hy 'vtt' (webvtt) (external)
     Subs  --sid=87 --slang=mi 'vtt' (webvtt) (external)
     Subs  --sid=88 --slang=am 'vtt' (webvtt) (external)
     Subs  --sid=89 --slang=en 'vtt' (webvtt) (external)
     Subs  --sid=90 --slang=jv 'vtt' (webvtt) (external)
     Subs  --sid=91 --slang=gd 'vtt' (webvtt) (external)
     Subs  --sid=92 --slang=lv 'vtt' (webvtt) (external)
     Subs  --sid=93 --slang=ms 'vtt' (webvtt) (external)
     Subs  --sid=94 --slang=mt 'vtt' (webvtt) (external)
 (+) Subs  --sid=95 --slang=fr 'vtt' (webvtt) (external)
     Subs  --sid=96 --slang=cy 'vtt' (webvtt) (external)
     Subs  --sid=97 --slang=mn 'vtt' (webvtt) (external)
     Subs  --sid=98 --slang=ha 'vtt' (webvtt) (external)
     Subs  --sid=99 --slang=hu 'vtt' (webvtt) (external)
     Subs  --sid=100 --slang=zh-Hans 'vtt' (webvtt) (external)
     Subs  --sid=101 --slang=ga 'vtt' (webvtt) (external)
     Subs  --sid=102 --slang=az 'vtt' (webvtt) (external)
     Subs  --sid=103 --slang=sn 'vtt' (webvtt) (external)
     Subs  --sid=104 --slang=mr 'vtt' (webvtt) (external)
AO: [pulse] 48000Hz stereo 2ch float
VO: [opengl] 426x240 yuv420p
Opening video filter: [yadif]
AV: 00:00:26 / 00:53:25 (0%) A-V:  0.000 Cache: 10s+15MB
```

I'm using ffmpeg v3.2 that was build with `--enable-gnutls`, `--enable-nonfree` and `--enable-librtmp` options.

Can you please have a look and see if it is possible to fix this ?

**Comment by Jj0YzL5nvJ** [comment]
Try with `mpv --ytdl-format=bestvideo+bestaudio[acodec!=opus] https://www.youtube.com/watch?v=kiQa3DiiVys` or only `--ytdl-format=best`.

Like I mentioned early, I use ffmpeg 3.2.4 compiled with librtmp and openssl options. #3288
With your video, I see a different error only when I quit: `[ffmpeg] https: Stream ends prematurely at 1474560, should be 44087769`
The source of this error is the "ytdl-format=251", an opus file... In my case, only this file in this video triggers the error.

Things apart, when you build ffmpeg with `--enable-gnutls`, you add `--enable-gcrypt` too, librtmp is used for openssl. I don't think librtmp and gnutls can work together.
Use `./configure --help` in ffmpeg for read the descriptions.

**Comment by wiiaboo** [comment]
Use `--ytdl-raw-options=sub-lang=<langcode>,write-sub=` to restrict how many subtitles you get from Youtube.

**Comment by sebma** [comment]
I don't have use these options, because this issue seems to be fixed on mpv v0.26+ffmpeg v7:3.3.3
